# **Ejercicios de Repaso**
## Comparación de Dos Grupos Independientes

Un grupo de investigación en desarrollo integral está interesado en trabajar la relación entre la motivación intrínseca y la autoeficacia académica. Para ello, encuestan a un grupo de estudiantes de cuarto a quinto de secundaria de un colegio privado en Jesús María.

In [ ]:
# @title Instalar y cargar librerías
install.packages(c("rstatix", "skimr", "coin"))

library(tidyverse)
library(rstatix)

Installing packages into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [ ]:
# @title Cargar conjunto de datos
library(haven)

df = read_sav("https://github.com/renatoparedes/INEE/raw/main/Ejercicios/Base%20-%20Ejercicios%20Comparaci%C3%B3n%20de%202%20grupos.sav")

#print("Ten en cuenta que la base de datos de este ejercicio se denomina 'df'")
#df %>% print(n = 10)

La base de datos se llama "df". Esta conformada por las siguientes variables:

|Variables|Nombre|Tipo|Etiquetas
|---|---|---|---
Caso|-|-|-
TipoDoc|Tipo de Docente|Cualitativa Nominal|Docente con clases ordinales (1), Docente con clases gamificadas (2)
NSE|Nive Socioeconómico|Cualitativa Ordinal|C (1) y A (2)
Salón|Salón de clases|Cualitativa Ordinal|4to (1) y 5to (2)
MoInt|Motivación Intrínseca|Cuantitativa Continua|-
AutoAc|Autoeficacia Aadémica|Cuantitativa Continua|-

In [ ]:
df %>% print(n = 10) # Veamos los primeros 10 casos

# A tibble: 640 × 6
    Caso TipoDoc                            NSE       Salón     MotInt AutoAc
   <dbl> <dbl+lbl>                          <dbl+lbl> <dbl+lbl>  <dbl>  <dbl>
 1     1 1 [Docente con clases ordinales]   1 [C]     1 [4to]       19     21
 2     2 2 [Docente con clases gamificadas] 1 [C]     1 [4to]       19     20
 3     3 1 [Docente con clases ordinales]   2 [A]     1 [4to]       21     22
 4     4 2 [Docente con clases gamificadas] 2 [A]     1 [4to]       20     21
 5     5 1 [Docente con clases ordinales]   1 [C]     2 [5to]       17     21
 6     6 2 [Docente con clases gamificadas] 1 [C]     2 [5to]       19     20
 7     7 1 [Docente con clases ordinales]   2 [A]     2 [5to]       18     21
 8     8 2 [Docente con clases gamificadas] 2 [A]     2 [5to]       21     20
 9     9 1 [Docente con clases ordinales]   1 [C]     1 [4to]       17     21
10    10 2 [Docente con clases gamificadas] 1 [C]     1 [4to]       20     22
# ℹ 630 more rows


## Ejercicio 1

A partir de la revisión bibliográfica, que da soporte a sus objetivos específicos, los integrantes del grupo de investigación creen que podría haber diferencias en el nivel de motivación intrínseca según las estrategias de dictado de los docentes. Así, creen que aquellos estudiantes que trabajen con un docente que fomenta el uso de actividades gamificadas en clase tendrán un mayor nivel de motivación intrínseca que aquellos que no cursan con estos docentes.

1. En primer lugar, tenemos que definir las variables de estudio.

Variables|Tipo|Grupos
---|---|---
Motivación Intrínseca|Cuantitativa|-
Estrategias de dictado|Cualitativa|Gamificado (G) y No Gamificado (NG)

2. Luego de entender las variables que tenemos en el estudio, necesitamos plantear las hipótesis.

$HT$: Se cree que aquellos estudiantes que trabajen con un docente que fomente el uso de actividades gamificadas en clase tendrán un mayor nivel de motivación intrínseca que aquellos que no cursan con estos docentes.

- $H_{0}: \mu_{G} = \mu_{NG}$
- $H_{1}: \mu_{G} > \mu_{NG} (unilateral)$

3. Ahora que tenemos estos elementos, necesitamos evaluar la normalidad a fin de determinar si vamos ha hacer una prueba t de Student o U de Mann-Whitney.

In [ ]:
df %>%
group_by(TipoDoc) %>%
shapiro_test(MotInt)

TipoDoc variable statistic p             
1 1       MotInt   0.95294   0.000000013294
2 2       MotInt   0.96658   0.000000983946

$Hipótesis\ de\ normalidad$

- $H_{0}: hay\ normalidad$
- $H_{1}: No\ hay\ normalidad$

**Regla de decisión: Si el valor *p* es menor a .05, se rechaza la H0.**

Todos los valores *p* son menores a .05, por lo que se rechaza la H0.

***Conclusión:***
A partir de la prueba de normalidad de Shapiro-Wilk, se observa que no existe una distribución normal tanto en el grupo de docentes que no gamifica su clase, $SW = .95, p < .001 $, como en el grupo donde se gamifica la clase, $SW = .97, p < .001 $. Por todo ello, se realizará la prueba no paramétrica para comparación de dos grupos de U de Mann-Whitney.

4. Luego de determinar la normalidad, pasamos a evaluar las diferencias entre los grupos por medio del análisis correspondiente.

In [ ]:
df %>%
  wilcox_test(MotInt ~ TipoDoc, paired = F)

.y.    group1 group2 n1  n2  statistic
1 MotInt 1      2      320 320 21346    
  p                                         
1 0.0000000000000000000000000000000000000271

**Dado que es un contraste unilateral. Tenemos que dividir entre 2 el valor *p*.** Dicho esto, dado que el valor *p* es tan pequeño podemos concluir lo siguiente .000/2 = .000.

Así, se observa que el valor *p* es menor que .05 por lo que se rechaza la H0 del ejercicio.

5. Ahora, es necesario determinar <u>*que grupo tiene el mayor nivel de la variable*</u>. Para esto, y dado que estamos en el contexto de un contraste paramétrico, observaremos las medianas.

In [ ]:
df %>%
  group_by(TipoDoc) %>%
  summarise(median = median(MotInt),
            IQR = IQR(MotInt))

TipoDoc median IQR
1 1       18.5   2  
2 2       20.0   3

Según las medianas, el grupo de mayor nivel es el 2 que, según lo mencionado previamente, sería el grupo que tuvo las clases gamificadas. Dado que esto va de acorde a lo planteado en la H1 podemos inferir que aceptamos esta hipótesis alterna.

6. Determinar la magnitud del efecto.

Si bien ya conocemos que existen diferencias y el grupo de mayor nivel, es importante poder determinar si estas diferencias son relevantes. Para el caso de la prueba no paramétrica de U de Mann-Whitney se debe usar el estadístico de $r\ de\ Rosenthal$ para evaluar la magnitud.

In [ ]:
df %>%
  wilcox_effsize(MotInt ~ TipoDoc, paired = F)

.y.    group1 group2 effsize n1  n2  magnitude
1 MotInt 1      2      0.51147 320 320 large

A partir del criterio para evaluar la $r\ de\ Rosenthal$ se observa que existe una magnitud de la diferencia grande.

7. Conclusión

A partir de la prueba U de Mann-Whitney, se observa que existen diferencias estadísticamente significas en el nivel de motivación intrínseca según las estrategias de dictado, $U = 21346, p < .001$. Específicamente, se encontró que el grupo cuyas clases eran gamificadas ($Mdn = 20$) tenían un mayor nivel de motivación que el grupo que no tenía clases gamificadas ($Mdn = 18.5$). La magnitud de esta diferencia es grande, $r\ de\ Rosenthal = .51$.

## Ejercicio 2

Paralelamente, la literatura no es muy clara en tanto a si la motivación intrínseca varía según el nivel socioeconómico. Si bien se han encontrado diferencias en algunos trabajos, estas diferencias parecen tenues y contradichas con estudios mas recientes. No obstante, el grupo de investigación desea indagar en sí habría algún tipo de diferencias en el grupo encuestado.

1. En primer lugar, tenemos que definir las variables de estudio.

Variables|Tipo|Grupos
---|---|---
Motivación Intrínseca|Cuantitativa|-
Nivel Socioeconómico|Cualitativa|A y C

2. Luego de entender las variables que tenemos en el estudio, necesitamos plantear las hipótesis.

$HT$: Se cree que habrían diferencias en el nivel de motivación intrínseca según el nivel socioeconómico.

- $H_{0}: \mu_{A} = \mu_{C}$
- $H_{1}: \mu_{A} \neq \mu_{C} (bilateral)$

3. Ahora que tenemos estos elementos, necesitamos evaluar la normalidad a fin de determinar si vamos ha hacer una prueba t de Student o U de Mann-Whitney.

In [ ]:
df %>%
  group_by(NSE) %>%
  shapiro_test(MotInt)

NSE variable statistic p            
1 1   MotInt   0.96503   0.00000057646
2 2   MotInt   0.95905   0.00000008229

$Hipótesis\ de\ normalidad$

- $H_{0}: hay\ normalidad$
- $H_{1}: No\ hay\ normalidad$

**Regla de decisión: Si el valor *p* es menor a .05, se rechaza la H0.**

Todos los valores *p* son menores a .05, por lo que se rechaza la H0.

***Conclusión:***
A partir de la prueba de normalidad de Shapiro-Wilk, se observa que no existe una distribución normal tanto en el grupo que está en el nivel socioeconómico C, $SW = .97, p < .001 $, como en el grupo que se encuentra en el nivel A, $SW = .96, p < .001 $. Por todo ello, se realizará la prueba no paramétrica para comparación de dos grupos de U de Mann-Whitney.

4. Luego de determinar la normalidad, pasamos a evaluar las diferencias entre los grupos por medio del análisis correspondiente.

In [ ]:
df %>%
  wilcox_test(MotInt ~ NSE, paired = F)

.y.    group1 group2 n1  n2  statistic p              
1 MotInt 1      2      320 320 35669     0.0000000000168

 **Dado que es un contraste bilateral. NO tenemos que dividir entre 2 el valor *p*.** Así, se observa que el valor *p* es menor que .05 por lo que se rechaza la H0 del ejercicio y se acepta la H1, de que existen diferencias entre los grupos.

5. Ahora, es necesario determinar <u>*que grupo tiene el mayor nivel de la variable*</u>. Para esto, y dado que estamos en el contexto de un contraste paramétrico, observaremos las medianas.

In [ ]:
df %>%
  group_by(NSE) %>%
  summarise(median = median(MotInt),
            IQR = IQR(MotInt))

NSE median IQR
1 1   19     2  
2 2   20     2

Según las medianas, el grupo de mayor nivel es el 2 que, según lo mencionado previamente, sería el grupo que pertenece al nivel socioeconómico A.

6. Determinar la magnitud del efecto.

Si bien ya conocemos que existen diferencias y el grupo de mayor nivel, es importante poder determinar si estas diferencias son relevantes. Para el caso de la prueba no paramétrica de U de Mann-Whitney se debe usar el estadístico de $r\ de\ Rosenthal$ para evaluar la magnitud.

In [ ]:
df %>%
  wilcox_effsize(MotInt ~ NSE, paired = F)

.y.    group1 group2 effsize n1  n2  magnitude
1 MotInt 1      2      0.26608 320 320 small

A partir del criterio para evaluar la $r\ de\ Rosenthal$ se observa que existe una magnitud de la diferencia pequeña.

7. Conclusión

A partir de la prueba U de Mann-Whitney, se observa que existen diferencias estadísticamente significas en el nivel de motivación intrínseca según el nivel socio económico, $U = 35669, p < .001$. Específicamente, se encontró que el grupo que pertence al nivel socio económico A ($Mdn = 20$) tiene un mayor nivel de motivación que el grupo que pertenece al nivel socio económico C ($Mdn = 19$). La magnitud de esta diferencia es pequeña, $r\ de\ Rosenthal = .27$.

## Ejercicio 3

Finalmente, creen que aquellos estudiantes que tienen un grado mayor en el colegio, tendrían un mejor nivel de motivación intríseca. Distintos estudios apuntan a que el éxito y el logro escolar va de la mano con una mayor interiorización de las tareas y trabajos realizados en el aula. De esta forma, el encontrarse en un grado mayor, los estudiantes estarían evidenciando cierto nivel de autonomía y motivación propia para realizar sus distintos trabajos.

1. En primer lugar, tenemos que definir las variables de estudio.

Variables|Tipo|Grupos
---|---|---
Motivación Intrínseca|Cuantitativa|-
Salón|Cualitativa|4to y 5to

2. Luego de entender las variables que tenemos en el estudio, necesitamos plantear las hipótesis.

$HT$: Se cree que aquellos estudiantes que se encuentran en un grado mayor mostrarían un mayor nivel de motivación intrínseca que aquellos que se encuentrnan en un grado menor.

- $H_{0}: \mu_{4to} = \mu_{5to}$
- $H_{1}: \mu_{4to} < \mu_{5to} (unilateral)$

3. Ahora que tenemos estos elementos, necesitamos evaluar la normalidad a fin de determinar si vamos ha hacer una prueba t de Student o U de Mann-Whitney.

In [ ]:
df %>%
  group_by(Salón) %>%
  shapiro_test(MotInt)

Salón variable statistic p               
1 1     MotInt   0.93656   0.00000000018516
2 2     MotInt   0.95652   0.00000003800312

$Hipótesis\ de\ normalidad$

- $H_{0}: hay\ normalidad$
- $H_{1}: No\ hay\ normalidad$

**Regla de decisión: Si el valor *p* es menor a .05, se rechaza la H0.**

Todos los valores *p* son menores a .05, por lo que se rechaza la H0.

***Conclusión:***
A partir de la prueba de normalidad de Shapiro-Wilk, se observa que no existe una distribución normal tanto en los estudiantes que pertenecen a 4to, $SW = .94, p < .001 $, como los que pertenecen a 5to de secundaria, $SW = .96, p < .001 $. Por todo ello, se realizará la prueba no paramétrica para comparación de dos grupos de U de Mann-Whitney.

4. Luego de determinar la normalidad, pasamos a evaluar las diferencias entre los grupos por medio del análisis correspondiente.

In [ ]:
df %>%
  wilcox_test(MotInt ~ Salón, paired = F)

.y.    group1 group2 n1  n2  statistic p     
1 MotInt 1      2      320 320 46688     0.0505

[1] 0.02525

**Dado que es un contraste unilateral. Tenemos que dividir entre 2 el valor *p*.** El resultado de dicha división sería el siguiente: .0505/2 = .0253.

Así, se observa que el valor *p* es menor que .05 por lo que se rechaza la H0 del ejercicio.

5. Ahora, es necesario determinar <u>*que grupo tiene el mayor nivel de la variable*</u>. Para esto, y dado que estamos en el contexto de un contraste paramétrico, observaremos las medianas.

In [ ]:
df %>%
  group_by(Salón) %>%
  summarise(median = median(MotInt),
            IQR = IQR(MotInt))

Salón median IQR
1 1     19.0   2  
2 2     19.5   4

Según las medianas, el grupo de mayor nivel es el de quinto de media. Dado que esto va de acorde a lo planteado en la H1 podemos inferir que aceptamos esta hipótesis alterna.

6. Determinar la magnitud del efecto.

Si bien ya conocemos que existen diferencias y el grupo de mayor nivel, es importante poder determinar si estas diferencias son relevantes. Para el caso de la prueba no paramétrica de U de Mann-Whitney se debe usar el estadístico de $r\ de\ Rosenthal$ para evaluar la magnitud.

In [ ]:
df %>%
  wilcox_effsize(MotInt ~ Salón, paired = F)

.y.    group1 group2 effsize n1  n2  magnitude
1 MotInt 1      2      0.0773  320 320 small

A partir del criterio para evaluar la $r\ de\ Rosenthal$ se observa que existe una magnitud de la diferencia pequeña.

7. Conclusión

A partir de la prueba U de Mann-Whitney, se observa que existen diferencias estadísticamente significas en el nivel de motivación intrínseca según las estrategias de dictado, $U = 46688, p < .001$. Específicamente, se encontró que los estudiantes de quinto de media ($Mdn = 19.5$) tienen un mayor nivel de motivación que el los estudiantes de cuarto de secundaria ($Mdn = 19$). La magnitud de esta diferencia es pequeña, $r\ de\ Rosenthal = .08$.